bioblend/galaxy/invocations/__init__.py

```diff
@@ -190,23 +192,36 @@ def rerun_invocation(self, invocation_id: str, inputs_update: Optional[dict] = N
           Default is ``False``, but when setting parameters for a subworkflow,
           ``True`` is required.
 
-        :rtype: dict
-        :return: A dict describing the new workflow invocation.
+        :rtype: dict if ``remap=False``, or list if ``remap=True``
+        :return: A dict describing the new workflow invocation, or a list of remapped jobs.
 
         .. note::
           This method can only be used with Galaxy ``release_21.01`` or later.
         """
+        if remap:
+            errored_jobs = self.gi.jobs.get_jobs(state='error', invocation_id=invocation_id)
+            remap_failures = 0
+            rerun_jobs = []
+            for job in errored_jobs:
+                try:
+                    job = self.gi.jobs.rerun_job(job['id'], remap=True)
+                    rerun_jobs.append(job)
+                except ValueError:
+                    # should not occur, jobs from an invocation should always be remappable
+                    remap_failures += 1
+            if remap_failures:
+                raise ValueError(f'remap was set to True, but {remap_failures} out of {len(errored_jobs)} errored jobs could not be remapped.')
+            return rerun_jobs
+
         invocation_details = self.show_invocation(invocation_id)
         workflow_id = invocation_details['workflow_id']
         inputs = invocation_details['inputs']
-        wf_params = invocation_details['input_step_parameters']
+        step_ids_to_indices = {step['workflow_step_id']: step_index for step_index, step in enumerate(invocation_details['steps'])}
+        inputs.update({step_ids_to_indices[param['workflow_step_id']]: param['parameter_value'] for _, param in invocation_details['input_step_parameters'].items()})
```

Comment:
@nsoranzo I found that the `params` option was not working correctly so I removed it (as params can also be submitted via `inputs`) - does this look like an okay alternative?

Reply:
Mmmh, what's not working for you? I notice two issues:
- `invocation_details['input_step_parameters']` seems to be never populated by Galaxy, so `params_update` should actually be just `params` as in `invoke_workflow()`
- `payload = {'inputs': inputs, 'params': wf_params}` should have been `payload = {'inputs': inputs, 'parameters': wf_params}`

Does that help?

Reply:
> invocation_details['input_step_parameters'] seems to be never populated by Galaxy, so params_update should actually be just params as in invoke_workflow()

It does, but only if the input parameter is labelled.

> payload = {'inputs': inputs, 'params': wf_params} should have been payload = {'inputs': inputs, 'parameters': wf_params}

Whoops, maybe that was it - I'll try that, thanks!

Reply:
@nsoranzo it seems that `params` doesn't work with runtime parameters, can you confirm if that's the case?

Reply:
Ping @mvdbeek

Reply:
At least I couldn't get it to work with a runtime parameter, and the documentation:
```
        The ``params`` dict should be specified as follows::

          {STEP_ID: PARAM_DICT, ...}

        where PARAM_DICT is::

          {PARAM_NAME: VALUE, ...}
```
doesn't really make sense for a runtime parameter.

Reply:
This seems to make it clear that runtime parameters belong in `inputs`, not `parameters`: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/workflow/run_request.py#L282-L285